### ISSUE TITLE
Downloadable products not working

### ISSUE THREAD

**Issue by neil-jennings** [magento/magento2#24317]
### Preconditions (*)

1. Magento 2.2.9
2. PHP 7.1

### Steps to reproduce (*)

1. Login as a customer
2. Add downloadable product to cart
3. Checkout and pay using PayPal Express Checkout
4. Visit download links in my account

### Expected result (*)

1. Download should be available

### Actual result (*)

1. Download is stuck in pending status

![download_issue](https://user-images.githubusercontent.com/16046700/63764967-f23e7900-c8bf-11e9-84fe-515009bec031.png)

**Comment by neil-jennings** [comment]
@magento give me 2.2.9 instance

**Comment by magento-engcom-team** [comment]
Hi @neil-jennings. Thank you for your request.
Unfortunately, I can only deploy instances for `2.3` release line

**Comment by neil-jennings** [comment]
Reproduced issue on vanilla 2.2.8 & 2.2.9 installs with sample data

**Comment by engcom-Bravo** [comment]
Hello @neil-jennings. Thanks for applying. Those downloadable links become available to download only after the order have been invoiced. If Your PayPal Express Checkout is configured to automatically invoice orders (e.g. Payment Action = Sale) the links should be available for download right after the order is placed. Otherwise You need first to invoice the order. Did You considere all of that when You ran the scenario?

**Comment by neil-jennings** [comment]
The PayPal payment action is "Sale" and the orders come through as Complete.
Setting it to "Authorization" and manually invoicing is a work around but not really ideal for normal orders.

**Comment by engcom-Bravo** [comment]
@neil-jennings, in our case when payment method is set to "Sale" everything works as expected. But that's of course about version 2.3-develop and not the earlier versions You  reported about.  Probably the issue have already been fixed. Would You please try to reproduce Your issue on the latest 2.3-develop version. Thank You.

**Comment by neil-jennings** [comment]
It's working on 2.3, it's just a 2.2 issue, any chance of a fix or is upgrading to the next major version the only way ?

**Comment by engcom-Bravo** [comment]
@neil-jennings, fixes are not provided for all previous versions. They are delivered with the first release version coming after they've been fixed in a develop version. So due to our workflow we are supposed to close this issue. Thanks for applying.

**Comment by wd7080** [comment]
I have the same problem on Magento 2.3.2. I tested this myself and I found out this:
1. Under my orders, the order appears as complete.
2. Under my downloadable products, the order remains pending and the customer cannot get a download link.

Below is all the related configurations I am currently using and it is in line with Magento 2 documentation.
![Btree1](https://user-images.githubusercontent.com/33989344/64463417-03378780-d115-11e9-8bb2-2b10cd90c91d.JPG)
![Btree2](https://user-images.githubusercontent.com/33989344/64463419-0599e180-d115-11e9-8b34-43b82196a730.JPG)
![Btree3](https://user-images.githubusercontent.com/33989344/64463421-06cb0e80-d115-11e9-8823-38c5ce6d7279.JPG)
I tested this using a credit card AND paypal. It did *not* work on both.
